returnn/hdf.py

```diff
@@ -349,3 +362,6 @@ def run(self, task_id):
             )
 
         out_hdf.close()
+
+        if len(excluded_segments):
+            self._write_text(self.out_excluded_segments[task_id - 1], excluded_segments)
```

Comment:
If you define a file as output, I would say it should always exist. If no segments are excluded, then maybe create an empty file?

Other suggestion: collect the segments of all subtasks and then only create one file with missing segments. (you would need another task for this, I guess)

Reply:
> collect the segments of all subtasks and then only create one file with missing segments.

I do not want to create too many accesses to the same file.

Reply:
I would also prefer not having files in `self.out_excluded_segments` that are not created. If subsequent jobs would use this, they might easily fail. Creating one joint file with all segments seems reasonable to me, it doesn't add much overhead and subsequent jobs would most likely use a joint segment list anyway.

Reply:
Just saying they might fail if they assume that all files in `self.out_excluded_segments` exist.

I would vote for a single file including all missing segments, although no strong feelings are involved on my side either :smile:

Reply:
> you should just check whether the list is empty or not

You would actually have to check whether to file exists or not right? I am not sure, but is it possible that there might be issues with sisyphus warning regarding missing files? e.g. only 3 and 10 exists, but you pass `out_excluded_segments` to a job (which actually correctly handles the list within the job), still I think there might be a warning. 
I would vote for a single file, which is empty if no segments are excluded.

Reply:
Ok, I will create one single file that is created also when there is no empty segment.